src/indexer/index_writer.rs

```diff
@@ -266,19 +278,17 @@ impl<D: Document> IndexWriter<D> {
     /// `TantivyError::InvalidArgument`
     pub(crate) fn new(
         index: &Index,
-        num_threads: usize,
-        memory_budget_in_bytes_per_thread: usize,
+        options: IndexWriterOptions,
```

Comment:
I think a reference is a tiny bit better here.

```suggestion
        options: &IndexWriterOptions,
```